packages/console/src/ds-components/Select/index.tsx

```diff
@@ -42,10 +45,19 @@ function Select<T extends string>({
   isClearable,
   size = 'large',
   dropdownPosition = 'bottom',
+  isSearchEnabled,
 }: Props<T>) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
   const [isOpen, setIsOpen] = useState(false);
+  const [searchInputValue, setSearchInputValue] = useState('');
+  const [width, setWidth] = useState(window.innerWidth);
   const anchorRef = useRef<HTMLInputElement>(null);
   const current = options.find((option) => value && option.value === value);
+  const filteredOptions = searchInputValue
+    ? options.filter(({ value }) =>
+        value.toLocaleLowerCase().includes(searchInputValue.toLocaleLowerCase())
+      )
+    : options;
```

Comment:
This would be better to be wrapped in a `useRef` hook

Reply:
@charIeszhao I used useState instead because it needs to trigger a re-render.

Reply:
Sorry, I was thinking about using `useMemo` but I wrote `useRef` by accident...